CorsixTH/Lua/entities/machine.lua

```diff
@@ -289,6 +289,19 @@ end
 
 function Machine:machineRepaired(room)
   room.needs_repair = nil
+  self:reduceStrengthOnRepair(room)
+  self:setRepairing(nil)
+  setSmoke(self, false)
+  local taskIndex = self.hospital:getIndexOfTask(self.tile_x, self.tile_y, "repairing")
+  self.hospital:removeHandymanTask(taskIndex, "repairing")
+end
+
+--! Calculates if machine strength should be reduced as a result of repair
+-- Also resets 'times_used' value
+--!param room (object) machine room
+function Machine:reduceStrengthOnRepair(room)
+  local minimum_possible_strength = 2
+  if self.strength <= minimum_possible_strength then return end
```

Comment:
Early return doesn't reset times used (very important :) ) 

Two options:
Add it to early return if block e.g.
```lua
  if self.strength <= minimum_possible_strength then
    self.times_used = 0
    return
  end
```
Or move `self.times_used = 0` at line 315 into the calling function `Machine:machineRepaired(room)` to happen after `self:reduceStrengthOnRepair(room)` at line 292.

Reply:
> Early return doesn't reset times used (very important :) )
> 
> Two options: Add it to early return if block e.g.
> 
> ```lua
>   if self.strength <= minimum_possible_strength then
>     self.times_used = 0
>     return
>   end
> ```
> 
> Or move `self.times_used = 0` at line 315 into the calling function `Machine:machineRepaired(room)` to happen after `self:reduceStrengthOnRepair(room)` at line 292.

Oh, my bad. 
That's really important. 
Should testing more carefully by me instead of being lazy. 
Thanks. 👍